json-rpc/types/src/views.rs

```diff
@@ -520,13 +520,11 @@ impl ScriptView {
 
 impl From<Transaction> for TransactionDataView {
     fn from(tx: Transaction) -> Self {
-        let x = match tx {
-            Transaction::BlockMetadata(t) => {
-                t.into_inner().map(|x| TransactionDataView::BlockMetadata {
-                    timestamp_usecs: x.1,
-                })
-            }
-            Transaction::GenesisTransaction(_) => Ok(TransactionDataView::WriteSet {}),
+        match tx {
+            Transaction::BlockMetadata(t) => TransactionDataView::BlockMetadata {
+                timestamp_usecs: t.timestamp_usec(),
+            },
+            Transaction::GenesisTransaction(_) => TransactionDataView::WriteSet {},
```

Comment:
should we rename this too?

Reply:
GenesisTransaction(WriteSetPayload),

I think keep it as WriteSet is better, GenesisTransaction seems too specific, although I don't mind to rename, but will need add new type to TransactionDataView, maybe another diff if we prefer.